x/execution/internal/keeper/keeper.go

```diff
@@ -221,14 +225,30 @@ func (k *Keeper) distributePriceShares(ctx sdk.Context, execAddress, runnerAddre
 	}
 
 	runnerCoins, _ := sdk.NewDecCoinsFromCoins(coins...).MulDecTruncate(runnerShare).TruncateDecimal()
-	serviceCoins := coins.Sub(runnerCoins)
-	// TODO: add emitters share
+	emittersCoins, _ := sdk.NewDecCoinsFromCoins(coins...).MulDecTruncate(emittersShare).TruncateDecimal()
+	serviceCoins := coins.Sub(runnerCoins).Sub(emittersCoins)
+
+	var voters []sdk.AccAddress
+	var emitterCoins sdk.Coins
+	if store.Has(voteKey(execHash)) {
+		k.cdc.MustUnmarshalBinaryLengthPrefixed(store.Get(voteKey(execHash)), &voters)
+		// divide coins for each emitter
+		emitterCoins, _ = sdk.NewDecCoinsFromCoins(emittersCoins...).QuoDec(sdk.NewDec(int64(len(voters)))).TruncateDecimal()
+	}
+
+	// if there is no emitter get rest coins to runner
+	if emitterCoins.IsZero() {
```

Comment:
why should we do that? why would we have no emitters in the first place? and why giving to the runner and not the service or get back to the user etc... we never talked about that...

Reply:
> why would we have no emitters in the first place?

See above. Also, it might happen that coins/len(emitters) == 0, so we can either pay only a few random emitters or none at all.

> and why giving to the runner

Because I took the reward from runners' share.

Reply:
> Because I took the reward from runners' share.

No, you're giving to the runner from the execution here... or I think there is a big confusion with what is the runner and all the different types of runners (emitters/executors).
The different types of runners will do a specific job and will be rewarded for that and not for the job of others.

Also, I'm seeing that in fact, this instruction is only when the coins are zero and you add zero to the existing coins. So yes we will actually not pay anyone and this whole instruction is useless isn't it?